contracts/tests/test_token_locker.cairo

```diff
@@ -0,0 +1,137 @@
+
+    start_prank(CheatTarget::One(token), owner);
+    token_dispatcher.approve(locker, 1000);
+    stop_prank(CheatTarget::One(token));
+
+    start_warp(CheatTarget::One(locker), 100);
+    start_prank(CheatTarget::One(locker), owner);
+    locker_dispatcher.lock(token, 1000);
+    stop_prank(CheatTarget::One(locker));
+
+    assert(token_dispatcher.balanceOf(owner) == 0, 'balanceOf owner not 0');
+
+    assert(token_dispatcher.balanceOf(locker) == 1000, 'balanceOf locker not 1000');
+
+    assert(
+        locker_dispatcher.get_locked_amount(token, owner, 100) == 1000,
+        'lockedAmount owner not 1000'
+    );
+
+    assert(locker_dispatcher.get_time_left(token, owner, 100) == 200, 'time left not 200');
+    stop_warp(CheatTarget::One(locker));
+}
+
+#[test]
+#[should_panic(expected: ('Still locked',))]
+fn test_unlock_early() {
+    let (owner, locker, token) = setup();
+    let token_dispatcher = IUnruggableMemecoinDispatcher { contract_address: token };
+    let locker_dispatcher = ITokenLockerDispatcher { contract_address: locker };
+
+    start_prank(CheatTarget::One(token), owner);
+    token_dispatcher.approve(locker, 1000);
+    stop_prank(CheatTarget::One(token));
+
+    start_warp(CheatTarget::One(locker), 100);
+    start_prank(CheatTarget::One(locker), owner);
+    locker_dispatcher.lock(token, 1000);
+    stop_prank(CheatTarget::One(locker));
+
+    assert(token_dispatcher.balanceOf(owner) == 0, 'balanceOf owner not 0');
+
+    start_warp(CheatTarget::One(locker), 200);
+    start_prank(CheatTarget::One(locker), owner);
+    locker_dispatcher.unlock(token, 100);
+    stop_prank(CheatTarget::One(locker));
+    stop_warp(CheatTarget::One(locker));
+}
+
+#[test]
+fn test_unlock() {
+    let (owner, locker, token) = setup();
+    let token_dispatcher = IUnruggableMemecoinDispatcher { contract_address: token };
+    let locker_dispatcher = ITokenLockerDispatcher { contract_address: locker };
+
+    start_prank(CheatTarget::One(token), owner);
+    token_dispatcher.approve(locker, 1000);
+    stop_prank(CheatTarget::One(token));
+
+    start_warp(CheatTarget::One(locker), 100);
+    start_prank(CheatTarget::One(locker), owner);
+    locker_dispatcher.lock(token, 1000);
+    stop_prank(CheatTarget::One(locker));
+
+    assert(token_dispatcher.balanceOf(owner) == 0, 'balanceOf owner not 0');
+
+    start_warp(CheatTarget::One(locker), 300);
+    start_prank(CheatTarget::One(locker), owner);
+    locker_dispatcher.unlock(token, 100);
+    stop_prank(CheatTarget::One(locker));
+
+    assert(token_dispatcher.balanceOf(locker) == 0, 'balanceOf locker not 0');
+
+    assert(token_dispatcher.balanceOf(owner) == 1000, 'balanceOf owner not 1000');
+
+    assert(locker_dispatcher.get_locked_amount(token, owner, 100) == 0, 'lockedAmount owner not 0');
+
+    assert(locker_dispatcher.get_time_left(token, owner, 100) == 0, 'time left not 0');
+    stop_warp(CheatTarget::One(locker));
+}
+
```

Comment:
add a `test_unlock_not_owner` to cover unlock calls by people who didn't lock

Reply:
done